### DISCUSSION TITLE
Ariston Alteas/Genus One+ boiler error reset possible ?

### DISCUSSION THREAD

**Discussion by dannyx94** [john30/ebusd#1008] [Q&A]
Hello. First of all, big thanks to everyone for your effort to make this project possible. I'm really interested in trying it myself, even though I know very little about code and stuff - good learning experience. I'm mostly interested in one simple task: resetting a "blocking" error in an Ariston Alteas (Genus) boiler. The Genus boiler is mentioned as tested at one point (I think by the author himself), so I was curious if someone has tried this feature. I have the Alteas model which is essentially the same thing AFAIK.

Why this would be a game-changer (for me at least): I'm away from home most of the time and if the gas gets shut off, the boiler throws a "501: No flame detected" error, which is normal. The problem is that it can only be reset locally from the control panel - there's no option to reset it via the app and even a power cycle of the unit doesn't get rid of it, so now the house is freezing for however long it takes me to get back, which can be days or even weeks on end.....not too smart there, Ariston. I can understand they're treating it as a safety hazard, but at least give me the option to take the risk...

Again, I'm a novice when it comes to programming, but if I understand correctly, in layman terms, the Ebus adapter is capable of listening in on the bus and reporting its activity to the RaspberryPi (as an example). The Alteas model which I have has also has a wired thermostat (Sensys HD) attached to the Ebus which also acts as a secondary display, mimicking the controls on the device's front face, including the option to reset errors. Assuming someone cracked the code and managed to read the info sent over the bus, I imagine it would be possible to send a reset command as well. I hope this is the case. I'm also hoping it wouldn't interfere with any existing devices on the bus, again, if someone tried it.

I don't want to dive too deep into this project just yet, so any input from any of you who experimented is good enough for now. Cheers !

**Comment by wrongisthenewright** [comment]
I've just seen your question, I was looking for something else and I stumbled upon it, Ysard has a Chaffoteaux boiler (Ariston Group) and he's been able to decode the error handling protocol, and even better the code to do the reset, take a look..

https://github.com/ysard/ebusd_configuration_chaffoteaux_bridgenet?tab=readme-ov-file#protocol-for-the-errors

**Reply by dannyx94** [reply]
Thanks for sharing that. I haven't actually set anything up, especially since I'm an absolute novice when it comes to programming, so this goes way over my head, but it's good to know it's there. I plan to take it really slow and set up the hardware first. I need to make sure the official Sensys thermostat/wired control panel still works in parallel with the Ebus adapter "hat" for the Raspberry Pi, since that's the platform I plan to use. I did some very minimal setup so far by installing the "hat" on the Raspberry Pi and trying out some commands without the hat connected to the Ebus, just to get a feel for how it works. It appears to work, so that's a good sign. I shall now attempt to connect the wires to the Ebus and see what happens...

**Reply by wrongisthenewright** [reply]
I have a complex system, with hybrid module and heat pump alongside the boiler, with Sensys and light gateway (a lot of traffic on the bus!!) and it works perfectly.

You should grab the bus in a spot where you can balance bus latency and wifi latency. In my case the best solution was to wire the adapter directly on the boiler bus port. 

The difficult part is the SW one. The Ebusd/adapter/ha integration is a little tricky but doable .

**Reply by dannyx94** [reply]
So do you have the Mira C or an Ariston ? Forgot to ask. I wonder how many of those "messages" the chap decoded are directly compatible with the Ariston.

**Reply by wrongisthenewright** [reply]
I've got the Ariston. Mira C is a model of the Chaffiteaux brand (Ariston group) . Both seems to use some of the same electronic

**Reply by dannyx94** [reply]
That's great news. I thought I'd be the only one struggling with an Ariston, since it seems it's most popular in Europe, but not anywhere else. I'm comparing the Mira's manual with Ariston's and most of the service menu sections do match up in terms of their 3-digit code. I don't know if it's relevant for the project at hand, but just as a bit of curiosity. The error codes in the manuals also seem to match to a degree - at least the codes that are common to both boilers, as some are found only on either one of the two. 501 for example is the same on both.

**Reply by dannyx94** [reply]
Approximately how many of the "messages" found by Ysard were directly compatible with your Ariston and how many did you have to adapt yourself to your model ?

**Reply by wrongisthenewright** [reply]
Given that some of Ysard messages came in some way from my repo 

https://github.com/wrongisthenewright/ebusd-configuration-ariston-bridgenet

I'd say that probably you'll get many values out of it.

My HVAC system is 3 years old so, if your boiler is newer, some codes may be missing or different.

**Reply by dannyx94** [reply]
Thanks. That info is invaluable to me ! I surmise you have a Genus boiler ? My Alteas boiler is roughly 1 year old as of its installation - not the actual release on the market, though it IS indeed the most recent model, so it's a 2022-23-ish model by my calculations, so it should behave the same as yours.

**Comment by dannyx94** [comment]
> You should grab the bus in a spot where you can balance bus latency and wifi latency. In my case the best solution was to wire the adapter directly on the boiler bus port.
>
The Alteas One has a built-in WiFi module that's connected to the main board through a separate connector - it does not require an external module connected to the bus for WiFi access, if that's what you are referring to here. Right now the only thing I have connected on the bus is the Sensys module, which also acts as my indoor thermostat. I'll run another wire from the ebus connector to the hat on the Pi and see what happens.

**Reply by wrongisthenewright** [reply]
Yoi have 2 options. Connect the hat 2 wires in parallel on the Sensys side (connect the 2 wires coming from the hat to the Sensys side connectors like prolonging the bus or you can connect the hat directly on the boiler PCB , always in parallel with the wires already present.

It depends on your house config, if the boiler is indoor and you have good wifi coverage in both places it should be indifferent where you tap the bus.
Please take care and do the wiring without ac power connected to the boiler.

**Reply by dannyx94** [reply]
Oh no, I won't be using WiFi on the Raspberry's side, if that's what you meant - wired ethernet instead. I'm all against WiFi to be honest :)

**Comment by dannyx94** [comment]
Quick question: if I open and edit any of the .csv files (for message definitions) in Excel, whenever I try to save them, I get a message about some "features getting lost if I continue"....does this affect the daemon when it loads the file ?

**Comment by wrongisthenewright** [comment]
You need to be sure that the CSV is comma separated values, without double quotes wrapping the various fileds

**Reply by dannyx94** [reply]
Not entirely sure how to check that, other than using Excel's "import data", which seems to delimit the columns correctly. The same thing happens if I just open the file directly by double-clicking, so I'm not sure if there's a difference.
EDIT: Ok, nevermind. I figured it out to a degree: you don't open CSVs in Excel directly - you import them OR edit them in notepad, just a lot harder to follow

**Reply by wrongisthenewright** [reply]
I use a normal text editor to customize my CSV so I don't know the correct procedure using XLS

**Reply by dannyx94** [reply]
I guess that works too, though it's tougher to follow. I just opened it in Notepad.

Should I expect the addresses of the boiler and Sensys to be different on my bus ?

**Reply by wrongisthenewright** [reply]
Can be but I don't think so. Ebusctl info run after some minutes after you have turned on wbusd should get you the exact list of the discovered devices.

Try and see what happens :)

Seriously you won't break anything. At worst if you have issues just disconnect the adapter and power cycle your boiler. It's quite resilient and I have done worst with my far busier bus ...

**Reply by dannyx94** [reply]
Will do. I guess there are no human-readable messages coming on the bus, so the only way you were able to figure out what each message actually meant was by trial and error and working out which message means what - like code XXXXXX means "heating on" and code YYYY means "hot water on". The bus is not going to TELL you this - it just spits out hex from what I understand. I'm always curious how people worked out each line in the .csv file !

**Comment by romicaiarca** [comment]
@dannyx94 , did you find a way to reset via ebusd Ariston boiler?

**Reply by dannyx94** [reply]
No, I haven't actually put in any real work into this project. I just got the EBUS adapter and did the most minimal setup, just enough to do a test run to make sure the boiler actually sends data to the RaspberryPi without affecting the existing SensysHD thermostat. I got some data coming in, so that part appears to work, but that's where I stopped for now due to lack of time and motivation, so I don't know how do use that data or what it all means - I have to move on to the next step and dig into those .csv files, which is pretty intimidating....

**Reply by romicaiarca** [reply]
ok, thank you! So we are pretty close each other. I managed to create a thermostat that looks like works in Home Assistant via ebus. But like you, I have some issues with my boiler and some times i get no flame error. So I'm looking for a way ti reset the error remotely. keep in touch if i find something.

**Reply by dannyx94** [reply]
Well you're way ahead of me, since you also got HomeAssistant running AND got the CSVs figured out, which is what I'm most interested in, as it was quite overwhelming. Were you able to find the list of message definitions for this particular model, or did you have to do some trial and error ?
